### PR TITLE
migrate legacy machine_id test to pytest

### LINKED ISSUE #827: Pytest: Migrate feature test `machine_id`
**What would you like to be added**:
The following test should be migrated to a pytest unit:
* [features/server/test/machine_id](https://github.com/gardenlinux/gardenlinux/blob/main/features/server/test/machine_id)

Thereby, there are already some pytest units ([here](https://github.com/gardenlinux/gardenlinux/tree/main/tests/helper/tests)), which can be used as blueprints for the migrated test.

**Why is this needed**:
We want to migrate each feature test to a corresponding pytest. Once we have done this, we can make the `chroot` and `kvm` platform tests the new default tests for Garden Linux builds.

### PR DESCRIPTION
**Pull request by chrinorse**
**How to categorize this PR?**

/kind enhancement
/area os
/os garden-linux

**What this PR does / why we need it**:
migrate legacy machine_id test to pytest
**Which issue(s) this PR fixes**:
Fixes #827 

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: features/server/test/test_machine_id.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from helper.tests.machine_id import machine_id
+import pytest
+
+@pytest.mark.skip_feature("server")
```
Comment (nanory) [inline features/server/test/test_machine_id.py:4]: Can we rename this mark to `skip_if_feature_missing`

File: features/server/test/test_machine_id.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+def test_machine_id(client):
```
Comment (nanory) [inline features/server/test/test_machine_id.py:5]: We should probably add a small comment section that describes this test.

File: tests/conftest.py [hunk 3]
```diff
@@ -334,6 +334,20 @@ def client(testconfig, iaas, imageurl, request) -> Iterator[RemoteClient]:
         raise ValueError(f"invalid {iaas=}")
 
 
+@pytest.fixture(autouse=True)
+def test_features(client, request):
+    (exit_code, output, error) = client.execute_command("cat /etc/os-release", quiet=True)
```
Comment (nanory) [inline tests/conftest.py:339]: We should add a documentation text to this fixture explaining what it does and why it is needed.

**Review by nanory — CHANGES_REQUESTED** [review]

File: tests/conftest.py [hunk 4]
```diff
@@ -5,6 +5,9 @@
 import sys
 import os
 import sys
+import mimetypes
+import tarfile
+import subprocess
```
Comment (nanory) [inline tests/conftest.py:10]: This imports are not needed anymore

File: tests/conftest.py [hunk 5]
```diff
@@ -334,6 +337,21 @@ def client(testconfig, iaas, imageurl, request) -> Iterator[RemoteClient]:
         raise ValueError(f"invalid {iaas=}")
 
 
+def pytest_collection_modifyitems(config, items):
```
Comment (nanory) [inline tests/conftest.py:340]: Add a small description, why this `pytest_collection_modifyitems` hook is needed.

File: tests/conftest.py [hunk 6]
```diff
@@ -334,6 +337,21 @@ def client(testconfig, iaas, imageurl, request) -> Iterator[RemoteClient]:
...
+    skip = pytest.mark.skip(reason="test is not part of the enabled features")
+    iaas = config.getoption("--iaas")
+    config_file = config.getoption("--configfile")
+    with open(config_file) as f:
```
Comment (nanory) [inline tests/conftest.py:344]: If the configuration file does not exists, we should fail.

**Review by nanory — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 5 above)
Comment (chrinorse) [reply tests/conftest.py:337]: Added a function documentation

**Review by nanory — APPROVED** [review]

**Pull request merged**